pkg/apis/monitoring/v1/types.go

```diff
@@ -880,3 +880,27 @@ const (
 	RoleEndpointSlice = "endpointslice"
 	RoleIngress       = "ingress"
 )
+
+// NativeHistogramConfig extends the native histogram configuration settings.
+// +k8s:openapi-gen=true
+type NativeHistogramConfig struct {
+	// Whether to scrape a classic histogram that is also exposed as a native histogram.
+	// It requires Prometheus >= v2.45.0.
+	//
+	// +optional
+	ScrapeClassicHistograms *bool `json:"scrapeClassicHistograms,omitempty"`
+
+	// If there are more than this many buckets in a native histogram,
+	// buckets will be merged to stay within the limit.
+	// It requires Prometheus >= v2.45.0.
+	//
+	// +optional
+	NativeHistogramBucketLimit *uint64 `json:"nativeHistogramBucketLimit,omitempty"`
+
+	// If the growth factor of one bucket to the next is smaller than this,
+	// buckets will be merged to increase the factor sufficiently.
+	// It requires Prometheus >= v2.50.0.
+	//
+	// +optional
+	NativeHistogramMinBucketFactor *float64 `json:"nativeHistogramMinBucketFactor,omitempty"`
```

Comment:
float fields are strongly discouraged because they can't be round-tripped (e.g. the returned value might be different from the value set by the client which could break reconcilers). The recommendation is to use `resource.Quantity` instead.

Reply:
Thanks @simonpasquier .
Change `*float64 -> *resource.Quantity` done.